src/main/java/io/appium/java_client/MultiTouchAction.java

```diff
@@ -62,16 +65,14 @@ public MultiTouchAction add(TouchAction action) {
      * Perform the multi-touch action on the mobile performsTouchActions.
      */
     public MultiTouchAction perform() {
-        int size = actions.build().size();
-        if (size > 1) {
+        List<TouchAction> touchActions = actions.build();
+        checkArgument(touchActions.size() > 0,
+                "MultiTouch action must have at least one TouchAction added before it can be performed");
+        if (touchActions.size() > 1) {
             performsTouchActions.performMultiTouchAction(this);
```

Comment:
you can `returns this` immediately here, so then `else` block is not needed

Reply:
@mykola-mokhnach 
I can't understand how to do it
```java
performsTouchActions.performMultiTouchAction(this);
```
is `void`

```java
performsTouchActions.performTouchAction(touchActions.get(0));
```
returns `TouchAction`

Reply:
```java
        if (touchActions.size() > 1) {
            performsTouchActions.performMultiTouchAction(this);
            resturn this;
        }
        //android doesn't like having multi-touch actions with only a single TouchAction...
        performsTouchActions.performTouchAction(touchActions.get(0));
        return this;
```